Automattic/WP-Job-Manager
# Introduce listing renewals

Fixes #848 
Based on discussion in https://github.com/Automattic/WP-Job-Manager/pull/2416

### Changes proposed in this Pull Request

* Adds the extend link to the actions when a listing is about to expire, extends the listing expiry date.
* Adds 'Expires soon' label to listings that are about to expire.
* Adds a new setting enabling or disabling extending a listing's expiry date.

### Testing instructions

* Set the expiry date of a job listing to a date within 5 days from now.
* Ensure the Job Manager pages and shortcodes are set up.
* In Job Submissions settings, enable extending job listings (Extend job listing expiry), it's turned off by default.
* Go to the Job Dashboard page. Notice the option to 'Extend' the job listing action.
* Click on 'Extend' and follow the steps. The new expiry date should be calculated based on the old expiry date, not the current date.


### New/Updated Hooks

* ~`job_manager_expiring_soon_days`~ `job_manager_renewal_days`: Number of days that determine if a listing is expiring soon.
* `renew_job_steps`: Fires when listing renewal steps have been added.

### Screenshot / Video

https://github.com/Automattic/WP-Job-Manager/assets/26160845/99a48604-dfb5-4b10-bc67-3f4cba96086c



### TODO

- ~Add new hooks for integrating with WC Paid Listings and Simple Paid Listings~ Done

### Update
*Extend* is now *Renew* which seems like a better description for the feature.
